code/game/machinery/requests_console.dm

```diff
@@ -137,6 +144,60 @@ var/list/obj/machinery/requests_console/requests_consoles = list()
 			if(!("[department]" in req_console_information))
 				req_console_information += department
 
+/obj/machinery/requests_console/proc/add_to_global_rc_list()
+	for(var/dept in master_department)
+		requests_consoles_categorised[dept] += src
+
+/obj/machinery/requests_console/proc/set_master_department(var/name)//this is fucking awful but less awful than updating all the maps and consoles
+	if(!isemptylist(master_department))
+		add_to_global_rc_list()
+		return "already setup"
+	var/list/command = list("Bridge","Captain's Desk","Chief Engineer's Desk","Chief Medical Officer's Desk","Head of Personnel's Desk","Head of Security's Desk","Research Director's Desk")
+	var/list/engineering = list("Atmospherics","Chief Engineer's Desk","Engineering","Mechanics")
+	var/list/medical = list("Chief Medical Officer's Desk","Genetics","Medbay","Chemistry","Virology")
+	var/list/research = list("Genetics","Research Director's Desk","Robotics","Science","Xenoarchaeology","Xenobiology","Mechanics")
+	var/list/security = list("Head of Security's Desk","Security")
+	var/list/service = list("Bar","Hydroponics","Kitchen","Head of Personnel's Desk")
+	var/list/cargo = list("Head of Personnel's Desk","Cargo Bay")
+	var/list/civillian = list("Pod Bay","Tool Storage","Chapel","EVA","Arrival shuttle","Locker Room","Janitorial","Head of Personnel's Desk")
+	for(var/subdept in command)
```

Comment:
if you really wanted to do this, why not just make a single associative list that associates each RC with its department?

For example
master departments = list(
Atmos = Engineering
Engineering = Engineering
Chief Engineer = Engineering)

then do a lookup
master_department = master_departments[name]

Reply:
consoles go to both command and their respective department, can't do it that way

Reply:
i did do it similarly to that until i had to also set the short form because of the switchboard which brought me back to the blocks of if()'s
the least sinful(and probably a couple milliseconds faster for server init) way would be to automatically save these into the map file but i have no idea how

Reply:
Since shorts are just department radio keys, you can almost certainly just pull those from the department radio keys list and clip off the : with an operation

Reply:
it seems you DO now sort them into an associative list so this is probably no longer necessary